iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/functional_interface.inl

```diff
@@ -131,10 +131,7 @@ inline Derived& AndThenWithValue<Derived, ValueType>::and_then(const Functor& ca
     if (*derivedThis)
     {
         auto callback = static_cast<and_then_callback_t>(callable);
-        if (callback)
```

Comment:
As this is a generic template I would propse to add an additional type trait `!is_convertiable<Functor, bool>`. What do you think?